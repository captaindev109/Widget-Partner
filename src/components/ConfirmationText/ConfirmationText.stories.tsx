import React from 'react';
import { Meta, Story } from '@storybook/react';
import styled from 'styled-components';
import { decorators } from '@/../.storybook/preview';
import { ConfirmationTextProps } from '@/types/types';
import Wrapper from '@/generics/Wrapper';
import ConfirmationText from '.';

// A wrapper to position the component
const StoryWrapper = styled(Wrapper)``;

// Name the folder to either generics or components
const meta: Meta = {
  title: 'Components/ConfirmationText',
  component: ConfirmationText,
  decorators,
};

export default meta;

// Showcasing the component
const Template: Story<ConfirmationTextProps> = (args) => (
  <StoryWrapper>
    <ConfirmationText {...args} />
    <ConfirmationText {...args} householdSize={4} />
  </StoryWrapper>
);

export const DefaultProductList = Template.bind({});

DefaultProductList.args = {
  householdSize: 1,
  email: 'test@hedvig.com',
  activationDate: '18/11/2022',
};
