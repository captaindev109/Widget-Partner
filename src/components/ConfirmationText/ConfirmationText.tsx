import { FC } from 'react';
import { ConfirmationTextProps } from '@/types/types';
import Card from '@/generics/Card';
import Text from '@/generics/Text';

//The main component with dynamic rendering and state
export const ConfirmationText: FC<ConfirmationTextProps> = ({ activationDate, householdSize, email }) => {
  const householdText = householdSize === 1 ? 'Du är nu skyddad' : `Du + ${householdSize - 1} är nu skyddade`;
  return (
    <Card>
      <Text
        variant="Body2"
        color="gray700"
      >{`${householdText}. Försäkringen aktiveras ${activationDate}. Ett bekräftelsemail har skickats till ${email}.`}</Text>
    </Card>
  );
};

export default ConfirmationText;
