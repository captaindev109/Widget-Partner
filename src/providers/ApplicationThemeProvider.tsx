import { ThemeProvider, DefaultTheme } from 'styled-components';
import GlobalStyles from '@/styles/Global';
import React from 'react';
import { ChildrenReturnType } from '@/types/types';

interface ApplicationThemeProviderProps {
  children?: ChildrenReturnType;
  theme: DefaultTheme;
}

const ApplicationThemeProvider = ({ children, theme }: ApplicationThemeProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default ApplicationThemeProvider;
