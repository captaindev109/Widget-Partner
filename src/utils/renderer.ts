import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
