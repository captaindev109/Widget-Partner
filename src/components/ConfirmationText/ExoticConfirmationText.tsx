import { FC } from 'react';
import ConfirmationText from '.';

const ExoticConfirmationText: FC = () => {
  // TODO - get user from store
  // TODO - get form data from store
  // const { userSelector, formDataSelector} = useStore();

  const { userSelector, formDataSelector } = {
    userSelector: {
      email: 'Test@hedvig.se',
    },
    formDataSelector: {
      householdSize: 2,
      activationDate: '18/11/2022',
    },
  };

  return <ConfirmationText {...userSelector} {...formDataSelector} />;
};

export default ExoticConfirmationText;
