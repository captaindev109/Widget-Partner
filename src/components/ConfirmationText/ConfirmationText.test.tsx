import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ApplicationThemeProvider from '@/providers/ApplicationThemeProvider';
import defaultTheme from '@/theme/theme';
import ConfirmationText from '.';

const MockComponent: React.FC = () => {
  return (
    <ApplicationThemeProvider theme={defaultTheme}>
      <ConfirmationText householdSize={1} email="test@hedvig.com" activationDate="18/11/2022" />
    </ApplicationThemeProvider>
  );
};

// describe function is a better way to organize our test cases and describe which component we are testing
describe('<ConfirmationText /> Test Cases', () => {
  // this test is a must for each component, any rendering issues with the component can be found
  test('it renders without crashing, all is well', () => {
    // render basically prints and exposes the component in nodejs environemt, it returns an object...
    render(<MockComponent />);
  });

  it('Snapshot test should pass', () => {
    const snap = renderer.create(<MockComponent />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
