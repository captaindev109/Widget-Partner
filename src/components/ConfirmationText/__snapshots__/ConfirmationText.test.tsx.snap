// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationText /> Test Cases Snapshot test should pass 1`] = `
<div
  className="Card-sc-fb64gk-0 leCJCh"
>
  <p
    className="Text__StyledText-sc-1jfsuh1-0 bvtSFa"
    color="gray700"
  >
    Du är nu skyddad. Försäkringen aktiveras 18/11/2022. Ett bekräftelsemail har skickats till test@hedvig.com.
  </p>
</div>
`;
